File: sdk/src/main/java/io/horizen/account/api/rpc/types/TxPoolInspect.java
package io.horizen.account.api.rpc.types;

import io.horizen.evm.Address;

import java.math.BigInteger;
import java.util.Map;

public class TxPoolInspect {
    public final Map<Address, Map<BigInteger, String>> pending;
    public final Map<Address, Map<BigInteger, String>> queued;

    public TxPoolInspect(
        Map<Address, Map<BigInteger, String>> pending,
        Map<Address, Map<BigInteger, String>> queued
    ) {
        this.pending = pending;
        this.queued = queued;
    }
}

File: sdk/src/main/scala/io/horizen/account/api/rpc/types/TxPoolInspect.scala
package io.horizen.account.api.rpc.types

import com.fasterxml.jackson.annotation.JsonView
import io.horizen.evm.Address
import io.horizen.json.Views

import java.math.BigInteger
import scala.collection.concurrent.TrieMap
import scala.collection.mutable

@JsonView(Array(classOf[Views.Default]))
class TxPoolInspect(
    val pending: TrieMap[Address, mutable.SortedMap[BigInteger, String]],
    val queued: TrieMap[Address, mutable.SortedMap[BigInteger, String]]) { }
